src/Avalonia.Controls/Navigation/INavigationRouter.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Avalonia.Controls
+{
+    public interface INavigationRouter : INotifyPropertyChanged
+    {
+        /// <summary>
+        /// If true, then <see cref="Back"/> or <see cref="Clear"/> will leave the <see cref="CurrentPage"/> null.
+        /// Otherwise there will always be a page left on the <see cref="CurrentPage"/>.
+        /// </summary>
+        bool AllowEmpty { get; set; }
+
+        /// <summary>
+        /// True if its possible to navigate backwards.
+        /// </summary>
+        bool CanGoBack { get; }
+
+        /// <summary>
+        /// The currently navigated page. Can be null if no navigation has taken place or
+        /// the user cleared (see <see cref="AllowEmpty"/>
+        /// </summary>
+        object? CurrentPage { get; }
+
+        /// <summary>
+        /// Navigates to a new page.
+        /// </summary>
+        /// <param name="destination">The destination page / url / viewmodel depending on the implementation.</param>
+        /// <param name="mode">How the navigation stack will behave when navigating.</param>
+        /// <returns>Task to await the navigation process.</returns>
+        Task NavigateTo(object? destination, NavigationMode mode = NavigationMode.Normal);
```

Comment:
I think it's best to add the Async suffix, so the developer knows he should use await to call it.

Reply:
`async` isn't valid for interface or abstract methods.

Reply:
I wasn't clear I meant the name. Rename Navigate to NavigateAsync.

Reply:
Ok. They've been renamed